packages/cli/src/util/update-notifier/update-worker.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * This file is spawned in the background and checks npm for the latest version
+ * of the CLI, then writes the version to the cache file.
+ *
+ * NOTE: Since this file runs asynchronously in the background, it's possible
+ * for multiple instances of this file to be running at the same time leading
+ * to a race condition where the most recent instance will overwrite the
+ * previous cache file resetting the `notified` flag and cause the update
+ * notification to appear for multiple consequetive commands. Not the end of
+ * the world, but something to be aware of.
+ */
+
+const fetch = require('node-fetch');
```

Comment:
Can we try `undici` here? As a company we are trying to start moving away from node-fetch and using the fetch included with Node.js (which is undici)

Reply:
I'm fine with switching to undici eventually, but that's out-of-scope for this PR.